### PR TITLE
Add missing author links in Beatmap Details

### LINKED ISSUE #7518: Mapper usernames should be hyperlinked consistently in song select, beatmap overlay
**Describe the new feature:** 

Mapper usernames should be hyperlinked consistently in all places including song select and beatmap overlay.

In direct, mapper is hyperlinked:
image

It is not in song select:
image

It is not in beatmap details overlay:
image

### PR DESCRIPTION
**Pull request by Denrage**
Closes #7518 

- [x] Depends on https://github.com/ppy/osu/pull/12503

SongSelect

https://user-images.githubusercontent.com/22132579/115258697-81ea0880-a131-11eb-873e-dc4256933fb8.mp4

BeatmapSetOverlay

https://user-images.githubusercontent.com/22132579/115258770-9201e800-a131-11eb-9551-fe9b7b651e1a.mp4

If there are other places where the author link is missing, let me know

### CODE REVIEW

**Comment by bdach** [comment]
[This has the exact same issue that the previous attempt at this had](https://github.com/ppy/osu/pull/7540#discussion_r376642332) in that the link doesn't change colour when hovered on the beatmap wedge.

File: osu.Game/Screens/Select/BeatmapInfoWedge.cs [hunk 1]
```diff
@@ -307,7 +308,7 @@ private void load(LocalisationManager localisation)
                                 Margin = new MarginPadding { Top = 10 },
                                 Direction = FillDirection.Horizontal,
                                 AutoSizeAxes = Axes.Both,
-                                Children = getMapper(metadata)
+                                Children = new[] { getMapper(metadata) },
```
Comment (bdach) [inline osu.Game/Screens/Select/BeatmapInfoWedge.cs:311]:
```suggestion
                                Child = getMapper(metadata),
```

File: osu.Game/Screens/Select/BeatmapInfoWedge.cs [hunk 2]
```diff
@@ -430,24 +431,20 @@ private void refreshBPMLabel()
                 ForceRedraw();
             }
 
-            private OsuSpriteText[] getMapper(BeatmapMetadata metadata)
+            private LinkFlowContainer getMapper(BeatmapMetadata metadata)
             {
-                if (string.IsNullOrEmpty(metadata.Author?.Username))
-                    return Array.Empty<OsuSpriteText>();
+                if (metadata.Author == null)
+                    return new LinkFlowContainer();
```
Comment (bdach) [inline osu.Game/Screens/Select/BeatmapInfoWedge.cs:437]:
```suggestion
                    return Empty();
```

\+ change return type of method to `Drawable`

**Review by bdach — CHANGES_REQUESTED** [review]
in addition to the above

**Comment by Denrage** [comment]
Does an issue exists which describes the goal splitting up the BufferedWedgeInfo? @bdach

**Comment by bdach** [comment]
I don't believe so, no. Not everything has to be a separate issue.

**Review by peppy — APPROVED** [review]

**Pull request merged**